torchvision/csrc/io/image/cpu/decode_jpeg.cpp

```diff
@@ -67,6 +67,61 @@ static void torch_jpeg_set_source_mgr(
   src->pub.next_input_byte = src->data;
 }
 
+inline void _convert_pixel_cmyk_to_rgb(
+    bool has_adobe_marker,
+    int c,
+    int m,
+    int y,
+    int k,
+    int& r,
+    int& g,
+    int& b) {
+  r = (has_adobe_marker) ? (k * c) / 255 : (255 - k) * (255 - c) / 255;
+  g = (has_adobe_marker) ? (k * m) / 255 : (255 - k) * (255 - m) / 255;
+  b = (has_adobe_marker) ? (k * y) / 255 : (255 - k) * (255 - y) / 255;
+}
+
+void convert_line_cmyk_to_rgb(
+    j_decompress_ptr cinfo,
+    const unsigned char* cmyk_line,
+    unsigned char* rgb_line) {
+  auto has_adobe_marker = cinfo->saw_Adobe_marker;
```

Comment:
I don't know if it's worth adding an image that has this marker in our tests, but did you check locally that this works OK?

Reply:
This is a good point, I have to check where the image from test is passing.

Reply:
I haven't found the usage of `has_adobe_marker` in Pillow or Opencv, so droping it. All the test data from ImageNet have it equal true. I also update the code to match exactly Pillow output.